diff --git a/tensorflow/lite/tools/cmake/modules/abseil-cpp.cmake b/tensorflow/lite/tools/cmake/modules/abseil-cpp.cmake
index b17f9b388d2..0c2969f06fa 100644
--- a/tensorflow/lite/tools/cmake/modules/abseil-cpp.cmake
+++ b/tensorflow/lite/tools/cmake/modules/abseil-cpp.cmake
@@ -22,7 +22,7 @@ include(OverridableFetchContent)
 
 OverridableFetchContent_Declare(
   abseil-cpp
-  GIT_REPOSITORY https://github.com/abseil/abseil-cpp
+  GIT_REPOSITORY git://github.com/abseil/abseil-cpp
   # Sync with tensorflow/third_party/absl/workspace.bzl
   GIT_TAG 997aaf3a28308eba1b9156aa35ab7bca9688e9f6
   GIT_SHALLOW TRUE
diff --git a/tensorflow/lite/tools/cmake/modules/clog.cmake b/tensorflow/lite/tools/cmake/modules/clog.cmake
index 93f10fb73c5..418688c5e83 100644
--- a/tensorflow/lite/tools/cmake/modules/clog.cmake
+++ b/tensorflow/lite/tools/cmake/modules/clog.cmake
@@ -21,7 +21,7 @@ include(OverridableFetchContent)
 
 OverridableFetchContent_Declare(
   clog
-  GIT_REPOSITORY https://github.com/pytorch/cpuinfo
+  GIT_REPOSITORY git://github.com/pytorch/cpuinfo
   # Sync with tensorflow/third_party/clog/workspace.bzl
   GIT_TAG d5e37adf1406cf899d7d9ec1d317c47506ccb970
   GIT_PROGRESS TRUE
diff --git a/tensorflow/lite/tools/cmake/modules/cpuinfo.cmake b/tensorflow/lite/tools/cmake/modules/cpuinfo.cmake
index 71f7f933828..a192780d5cf 100644
--- a/tensorflow/lite/tools/cmake/modules/cpuinfo.cmake
+++ b/tensorflow/lite/tools/cmake/modules/cpuinfo.cmake
@@ -21,7 +21,7 @@ include(OverridableFetchContent)
 
 OverridableFetchContent_Declare(
   cpuinfo
-  GIT_REPOSITORY https://github.com/pytorch/cpuinfo
+  GIT_REPOSITORY git://github.com/pytorch/cpuinfo
   # Sync with tensorflow/third_party/cpuinfo/workspace.bzl
   GIT_TAG 5916273f79a21551890fd3d56fc5375a78d1598d
   GIT_PROGRESS TRUE
diff --git a/tensorflow/lite/tools/cmake/modules/egl_headers.cmake b/tensorflow/lite/tools/cmake/modules/egl_headers.cmake
index 400f4fc779d..bf63340f4b3 100644
--- a/tensorflow/lite/tools/cmake/modules/egl_headers.cmake
+++ b/tensorflow/lite/tools/cmake/modules/egl_headers.cmake
@@ -21,7 +21,7 @@ include(FetchContent)
 
 OverridableFetchContent_Declare(
   egl_headers
-  GIT_REPOSITORY https://github.com/KhronosGroup/EGL-Registry.git
+  GIT_REPOSITORY git://github.com/KhronosGroup/EGL-Registry.git
   # No reference in TensorFlow Bazel rule since it's used for GPU Delegate
   # build without using Android NDK.
   GIT_TAG 649981109e263b737e7735933c90626c29a306f2
diff --git a/tensorflow/lite/tools/cmake/modules/eigen.cmake b/tensorflow/lite/tools/cmake/modules/eigen.cmake
index ed1ba25e690..2a47223e9e8 100644
--- a/tensorflow/lite/tools/cmake/modules/eigen.cmake
+++ b/tensorflow/lite/tools/cmake/modules/eigen.cmake
@@ -21,16 +21,18 @@ include(OverridableFetchContent)
 
 OverridableFetchContent_Declare(
   eigen
-  GIT_REPOSITORY https://gitlab.com/libeigen/eigen.git
+  URL https://gitlab.com/libeigen/eigen/-/archive/085c2fc5d53f391afcccce21c45e15f61c827ab1/eigen-085c2fc5d53f391afcccce21c45e15f61c827ab1.tar.gz
+  URL_HASH SHA256=cd72f0a56a95d85cb8a0160f4adc7fea72da49fbb7351ebb31c4e67e1a5fc8bd
+  # GIT_REPOSITORY https://gitlab.com/libeigen/eigen.git
   # Sync with tensorflow/third_party/eigen3/workspace.bzl
-  GIT_TAG 085c2fc5d53f391afcccce21c45e15f61c827ab1
+  # GIT_TAG 085c2fc5d53f391afcccce21c45e15f61c827ab1
   # It's not currently (cmake 3.17) possible to shallow clone with a GIT TAG
   # as cmake attempts to git checkout the commit hash after the clone
   # which doesn't work as it's a shallow clone hence a different commit hash.
   # https://gitlab.kitware.com/cmake/cmake/-/issues/17770
   # GIT_SHALLOW TRUE
-  GIT_PROGRESS TRUE
-  PREFIX "${CMAKE_BINARY_DIR}"
+  # GIT_PROGRESS TRUE
+  # PREFIX "${CMAKE_BINARY_DIR}"
   SOURCE_DIR "${CMAKE_BINARY_DIR}/eigen"
   LICENSE_FILE "COPYING.MPL2"
 )
diff --git a/tensorflow/lite/tools/cmake/modules/farmhash.cmake b/tensorflow/lite/tools/cmake/modules/farmhash.cmake
index 7a5160fa7eb..bfb6473fbcf 100644
--- a/tensorflow/lite/tools/cmake/modules/farmhash.cmake
+++ b/tensorflow/lite/tools/cmake/modules/farmhash.cmake
@@ -21,7 +21,7 @@ include(OverridableFetchContent)
 
 OverridableFetchContent_Declare(
   farmhash
-  GIT_REPOSITORY https://github.com/google/farmhash
+  GIT_REPOSITORY git://github.com/google/farmhash
   # Sync with tensorflow/third_party/farmhash/workspace.bzl
   GIT_TAG 0d859a811870d10f53a594927d0d0b97573ad06d
   # It's not currently possible to shallow clone with a GIT TAG
diff --git a/tensorflow/lite/tools/cmake/modules/flatbuffers.cmake b/tensorflow/lite/tools/cmake/modules/flatbuffers.cmake
index a3d01fd962d..d4d879542e2 100644
--- a/tensorflow/lite/tools/cmake/modules/flatbuffers.cmake
+++ b/tensorflow/lite/tools/cmake/modules/flatbuffers.cmake
@@ -21,7 +21,7 @@ include(FetchContent)
 
 OverridableFetchContent_Declare(
   flatbuffers
-  GIT_REPOSITORY https://github.com/google/flatbuffers
+  GIT_REPOSITORY git://github.com/google/flatbuffers
   # Sync with tensorflow/third_party/flatbuffers/workspace.bzl
   GIT_TAG v1.12.0
   GIT_SHALLOW TRUE
diff --git a/tensorflow/lite/tools/cmake/modules/fp16_headers.cmake b/tensorflow/lite/tools/cmake/modules/fp16_headers.cmake
index 0d039c01328..d975bd4f86c 100644
--- a/tensorflow/lite/tools/cmake/modules/fp16_headers.cmake
+++ b/tensorflow/lite/tools/cmake/modules/fp16_headers.cmake
@@ -22,7 +22,7 @@ include(FetchContent)
 
 OverridableFetchContent_Declare(
   fp16_headers
-  GIT_REPOSITORY https://github.com/Maratyszcza/FP16
+  GIT_REPOSITORY git://github.com/Maratyszcza/FP16
   # Sync with https://github.com/google/XNNPACK/blob/master/cmake/DownloadFP16.cmake
   GIT_TAG 0a92994d729ff76a58f692d3028ca1b64b145d91
   GIT_PROGRESS TRUE
diff --git a/tensorflow/lite/tools/cmake/modules/gemmlowp.cmake b/tensorflow/lite/tools/cmake/modules/gemmlowp.cmake
index da31e3c3fb8..6659bc4d83d 100644
--- a/tensorflow/lite/tools/cmake/modules/gemmlowp.cmake
+++ b/tensorflow/lite/tools/cmake/modules/gemmlowp.cmake
@@ -21,7 +21,7 @@ include(OverridableFetchContent)
 
 OverridableFetchContent_Declare(
   gemmlowp
-  GIT_REPOSITORY https://github.com/google/gemmlowp
+  GIT_REPOSITORY git://github.com/google/gemmlowp
   # Sync with tensorflow/third_party/gemmlowp/workspace.bzl
   GIT_TAG fda83bdc38b118cc6b56753bd540caa49e570745
   # It's not currently (cmake 3.17) possible to shallow clone with a GIT TAG
diff --git a/tensorflow/lite/tools/cmake/modules/opencl_headers.cmake b/tensorflow/lite/tools/cmake/modules/opencl_headers.cmake
index 8afbe444dc6..19ff92281cc 100644
--- a/tensorflow/lite/tools/cmake/modules/opencl_headers.cmake
+++ b/tensorflow/lite/tools/cmake/modules/opencl_headers.cmake
@@ -21,7 +21,7 @@ include(FetchContent)
 
 OverridableFetchContent_Declare(
   opencl_headers
-  GIT_REPOSITORY https://github.com/KhronosGroup/OpenCL-Headers
+  GIT_REPOSITORY git://github.com/KhronosGroup/OpenCL-Headers
   # Sync with tensorflow/third_party/opencl_headers/workspace.bzl
   GIT_TAG 0d5f18c6e7196863bc1557a693f1509adfcee056
   GIT_PROGRESS TRUE
diff --git a/tensorflow/lite/tools/cmake/modules/opengl_headers.cmake b/tensorflow/lite/tools/cmake/modules/opengl_headers.cmake
index c6dd7b842be..ebc1425728c 100644
--- a/tensorflow/lite/tools/cmake/modules/opengl_headers.cmake
+++ b/tensorflow/lite/tools/cmake/modules/opengl_headers.cmake
@@ -21,7 +21,7 @@ include(FetchContent)
 
 OverridableFetchContent_Declare(
   opengl_headers
-  GIT_REPOSITORY https://github.com/KhronosGroup/OpenGL-Registry.git
+  GIT_REPOSITORY git://github.com/KhronosGroup/OpenGL-Registry.git
   # No reference in TensorFlow Bazel rule since it's used for GPU Delegate
   # build without using Android NDK.
   GIT_TAG 0cb0880d91581d34f96899c86fc1bf35627b4b81
diff --git a/tensorflow/lite/tools/cmake/modules/ruy.cmake b/tensorflow/lite/tools/cmake/modules/ruy.cmake
index 752f527cf1b..2076f44c44f 100644
--- a/tensorflow/lite/tools/cmake/modules/ruy.cmake
+++ b/tensorflow/lite/tools/cmake/modules/ruy.cmake
@@ -21,10 +21,12 @@ include(OverridableFetchContent)
 
 OverridableFetchContent_Declare(
   ruy
-  GIT_REPOSITORY https://github.com/google/ruy
+  URL https://github.com/google/ruy/archive/e6c1b8dc8a8b00ee74e7268aac8b18d7260ab1ce.zip
+  # GIT_REPOSITORY https://github.com/google/ruy
   # Sync with tensorflow/third_party/ruy/workspace.bzl
-  GIT_TAG e6c1b8dc8a8b00ee74e7268aac8b18d7260ab1ce
-  GIT_PROGRESS TRUE
+  # GIT_TAG e6c1b8dc8a8b00ee74e7268aac8b18d7260ab1ce
+  # GIT_PROGRESS TRUE
+  URL_HASH SHA256=fa9a0b9041095817bc3533f7b125c3b4044570c0b3ee6c436d2d29dae001c06b
   SOURCE_DIR "${CMAKE_BINARY_DIR}/ruy"
 )
 OverridableFetchContent_GetProperties(ruy)
diff --git a/tensorflow/lite/tools/cmake/modules/vulkan_headers.cmake b/tensorflow/lite/tools/cmake/modules/vulkan_headers.cmake
index 7e1335a147c..0eff5b1f0d2 100644
--- a/tensorflow/lite/tools/cmake/modules/vulkan_headers.cmake
+++ b/tensorflow/lite/tools/cmake/modules/vulkan_headers.cmake
@@ -21,7 +21,7 @@ include(FetchContent)
 
 OverridableFetchContent_Declare(
   vulkan_headers
-  GIT_REPOSITORY https://github.com/KhronosGroup/Vulkan-Headers
+  GIT_REPOSITORY git://github.com/KhronosGroup/Vulkan-Headers
   # Sync with tensorflow/third_party/vulkan_headers/workspace.bzl
   GIT_TAG ec2db85225ab410bc6829251bef6c578aaed5868
   GIT_PROGRESS TRUE
diff --git a/tensorflow/lite/tools/cmake/modules/xnnpack.cmake b/tensorflow/lite/tools/cmake/modules/xnnpack.cmake
index e6cd5fd7482..bfa1f777d61 100644
--- a/tensorflow/lite/tools/cmake/modules/xnnpack.cmake
+++ b/tensorflow/lite/tools/cmake/modules/xnnpack.cmake
@@ -21,7 +21,7 @@ include(FetchContent)
 
 OverridableFetchContent_Declare(
   xnnpack
-  GIT_REPOSITORY https://github.com/google/XNNPACK
+  GIT_REPOSITORY git://github.com/google/XNNPACK
   # Sync with tensorflow/workspace2.bzl
   GIT_TAG 113092317754c7dea47bfb3cb49c4f59c3c1fa10
   GIT_PROGRESS TRUE
