diff --git a/tensorflow/lite/tools/cmake/modules/xnnpack.cmake b/tensorflow/lite/tools/cmake/modules/xnnpack.cmake
index eeb86758d63..ba133f9b71c 100644
--- a/tensorflow/lite/tools/cmake/modules/xnnpack.cmake
+++ b/tensorflow/lite/tools/cmake/modules/xnnpack.cmake
@@ -38,6 +38,9 @@ endif()
 set(XNNPACK_BUILD_TESTS OFF CACHE BOOL "Disable XNNPACK test.")
 set(XNNPACK_BUILD_BENCHMARKS OFF CACHE BOOL "Disable XNNPACK benchmarks.")
 
+# https://twitter.com/MaratDukhan/status/1632582746823446529
+set(XNNPACK_ENABLE_RISCV_VECTOR OFF CACHE BOOL "Disable Build XNNPACK with RISC-V Vector micro-kernels.")
+
 # The following line adds project of PTHREADPOOL, FP16 and XNNPACK which are
 # needed to compile XNNPACK delegate of TFLite.
 # Note, we introduce an intermediate subdirectory, see ${TFLITE_SOURCE_DIR}/tools/cmake/modules/xnnpack/CMakeLists.txt
