diff --git a/tensorflow/lite/tools/make/download_dependencies.sh b/tensorflow/lite/tools/make/download_dependencies.sh
index 0ab8307b07a..6dab676f1b9 100755
--- a/tensorflow/lite/tools/make/download_dependencies.sh
+++ b/tensorflow/lite/tools/make/download_dependencies.sh
@@ -77,7 +77,7 @@ download_and_extract() {
   rm -rf ${dir}/*  # Delete existing files.
   tempdir=$(mktemp -d)
   filepath="${tempdir}/$(basename ${url})"
-  curl -Lo ${filepath} ${url}
+  wget -q ${url} -O ${filepath}
   if [ -n "${sha256}" ]; then
     echo "checking sha256 of ${dir}"
     echo "${sha256}  ${filepath}" | sha256sum -c
