diff --git a/port/default/port_from_tf/logging.cc b/port/default/port_from_tf/logging.cc
index cedfaa3..e9dfeab 100644
--- a/port/default/port_from_tf/logging.cc
+++ b/port/default/port_from_tf/logging.cc
@@ -168,7 +168,7 @@ namespace darwinn {
 namespace internal {
 namespace {
 // TODO: can we make the logging level somehow local to a request?
-int log_level = 0;
+int log_level = 10;
 }  // namespace
 
 void SetLoggingLevel(int level) { log_level = level; }
