Upstream-Status: Pending

diff --git a/tensorflow/lite/examples/label_image/CMakeLists.txt b/tensorflow/lite/examples/label_image/CMakeLists.txt
index 545ceec2381..877f02b4b65 100644
--- a/tensorflow/lite/examples/label_image/CMakeLists.txt
+++ b/tensorflow/lite/examples/label_image/CMakeLists.txt
@@ -93,4 +93,5 @@ target_link_libraries(label_image
   tensorflow-lite
   profiling_info_proto
   protobuf
+  atomic
 )
